### PR TITLE
[IMP] Allow to filter by applications or l10n modules 

### PR DESCRIPTION
**Pull request by mreficent**
Superseeds https://github.com/OCA/maintainer-quality-tools/pull/535.

### CODE REVIEW

**Comment by mreficent** [comment]
Now it's green

Should I add tests to check the improvements?

File: travis/run_pylint.py [hunk 1]
```diff
@@ -218,28 +218,39 @@ def get_count_fails(linter_stats, msgs_no_count=None):
...
-                [os.path.join(path, item)
-                 for item in os.listdir(path)
-                 if os.path.isfile(os.path.join(path, item, '__init__.py')) and
-                 (not getaddons.is_module(os.path.join(path, item)) or
-                  getaddons.is_installable_module(os.path.join(path, item)))])
-        else:
-            if not getaddons.is_module(path) or \
-                    getaddons.is_installable_module(path):
-                subpaths.append(path)
+        if depth > 0:
```
Comment (moylop260) [inline travis/run_pylint.py]: What about `if depth <= 0: continue`?
In order to reduce the nested code and see a diff cleaner

File: travis/getaddons.py [hunk 2]
```diff
@@ -102,6 +104,57 @@ def get_modules_changed(path, ref='HEAD'):
     return modules_changed_path
 
 
+def get_dependencies(modules, module_name):
```
Comment (moylop260) [inline travis/getaddons.py]: Add docstring, please

File: travis/getaddons.py [hunk 3]
```diff
@@ -102,6 +104,57 @@ def get_modules_changed(path, ref='HEAD'):
...
+    result = set()
+    for dependency in modules.get(module_name, {}).get('depends', []):
+        result |= get_dependencies(modules, dependency)
+    return result | set([module_name])
+
+
+def get_dependents(modules, module_name):
+    result = set()
```
Comment (moylop260) [inline travis/getaddons.py]: Add docstring

File: travis/getaddons.py [hunk 4]
```diff
@@ -102,6 +104,57 @@ def get_modules_changed(path, ref='HEAD'):
...
+    return result | set([module_name])
+
+
+def add_auto_install(modules, to_install):
+    """ Append automatically installed glue modules to to_install if their
+    dependencies are already present. to_install is a set. """
+    found = True
+    while found:
+        found = False
+        for module in modules.keys():
```
Comment (moylop260) [inline travis/getaddons.py]: <s>Remove `.keys()`</s>
<s>Use `for module in modules` instead.</s>

In fact, you are using the key to get value then you could use:
`for module, module_data in dicc.items():`

File: travis/run_pylint.py (hunk 1 above)
Comment (mreficent) [reply travis/run_pylint.py]: What is faster, "not depth > 0" or "depth <= 0"?

Comment (moylop260) [reply travis/run_pylint.py]: pylint recommend us avoid the `not`:
 - screen shot 2018-04-11 at 14 48 53

Comment (moylop260) [reply travis/run_pylint.py]: FYI the `timeit` test show us that both sentences spent the same time:

- screen shot 2018-04-11 at 14 52 36

Comment (mreficent) [reply travis/run_pylint.py]: removed the not

**Comment by moylop260** [comment]
@StefanRijnhart Are you agree?

**Comment by mreficent** [comment]
Full :green_apple: :green_book: :green_heart: 

I could make green the coveralls, but that would imply to add a module test named l10n_something, but I think that it's unnecessary.

**Comment by StefanRijnhart** [comment]
I updated my PR and there are still some errors in getmodules.py https://github.com/OCA/OCB/pull/722

**Comment by mreficent** [comment]
@StefanRijnhart solved, try again 😁

**Comment by StefanRijnhart** [comment]
I restarted the build of the OCB PR

**Comment by StefanRijnhart** [comment]
The tests run with the module selection as expected!

**Comment by mreficent** [comment]
Squashed.

**Pull request merged**